From 5b103f9bb37e1c70dd801deb2eace8ffec314c2e Mon Sep 17 00:00:00 2001
Message-Id: <5b103f9bb37e1c70dd801deb2eace8ffec314c2e.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Daisy Sun <daisy.sun@intel.com>
Date: Mon, 5 May 2014 18:58:49 -0700
Subject: [PATCH 068/312] MUST_REBASE [VPG]: drm/i915: DPST debugfs read from
 reg varible

For some product with multiple sets of DPST register, the debugfs
result might be wrong.
Change DPST debugfs to read the register according to DPST structure
instead of fixed address.
Due to DPST IP concerns, this patch cannot be upstreamed
Issue: APDEV-2369

Signed-off-by: Daisy Sun <daisy.sun@intel.com>
Reviewed-on: https://android.intel.com/187407
Reviewed-by: Azad, Vinit <vinit.azad@intel.com>
Reviewed-by: Bloomfield, Jon <jon.bloomfield@intel.com>

Pulled from ABT/kernel-internal to GMin/gmin.

Change-Id: I0eb236fe4ade4065c152ee42952f10e9a207cc77
Signed-off-by: Oscar Mateo <oscar.mateo@intel.com>
---
 drivers/gpu/drm/i915/i915_debugfs.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index 30f9e5c..44a828d 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -1702,8 +1702,8 @@ static int i915_dpst_status(struct seq_file *m, void *unused)
 
 	mutex_lock(&dev_priv->dpst.ioctl_lock);
 
-	blm_hist_ctl = I915_READ(BLM_HIST_CTL);
-	blm_hist_guard = I915_READ(BLM_HIST_GUARD);
+	blm_hist_ctl = I915_READ(dev_priv->dpst.reg.blm_hist_ctl);
+	blm_hist_guard = I915_READ(dev_priv->dpst.reg.blm_hist_guard);
 
 	seq_printf(m, "histogram logic: %s\n",
 		   blm_hist_ctl & IE_HISTOGRAM_ENABLE ?
@@ -1728,7 +1728,7 @@ static int i915_dpst_status(struct seq_file *m, void *unused)
 	for (i = 0; i < DPST_DIET_ENTRY_COUNT; i++) {
 		if (i % columns == 0)
 			seq_printf(m, "\nbins %02d-%02d:", i, i + columns - 1);
-		seq_printf(m, "%10x", I915_READ(BLM_HIST_BIN));
+		seq_printf(m, "%10x", I915_READ(dev_priv->dpst.reg.blm_hist_bin));
 	}
 	seq_puts(m, "\n");
 
-- 
1.7.9.5

